openpolicedata/data_loaders.py

```diff
@@ -104,6 +104,61 @@ def load_csv(url, date_field=None, year_filter=None, agency_field=None, agency=N
         agency_field=agency_field, agency=agency)
 
     return table
+def load_excel(url, date_field=None, year_filter=None, agency_field=None, agency=None, limit=None, pbar=True):
+    '''Download XLS file to pandas DataFrame
+    
+    Parameters
+    ----------
+    url : str
+        Download URL for Excel
+    date_field : str
+        (Optional) Name of the column that contains the date
+    year_filter : int, list
+        (Optional) Either the year or the year range [first_year, last_year] for the data that is being requested. None value returns data for all years.
+    agency_field : str
+        (Optional) Name of the column that contains the agency name (i.e. name of the police departments)
+    agency : str
+        (Optional) Name of the agency to filter for. None value returns data for all agencies.
+    limit : int
+        (Optional) Only returns the first limit rows of the CSV
+    pbar : bool
+        (Optional) This progress bar argument is currently ignored. 
+        
+    Returns
+    -------
+    pandas DataFrame
+        DataFrame containing table imported from Excel spreadsheet
+    '''
+    if ".zip" in url:
+        with warnings.catch_warnings():
+            # Perhaps use requests iter_content/iter_lines as below to read large CSVs so progress can be shown
+            warnings.simplefilter("ignore", category=pd.errors.DtypeWarning)
+            try:
+                table = pd.read_excel(url, engine='openpyxl')
+            except urllib.error.HTTPError as e:
+                raise OPD_DataUnavailableError(*e.args, _url_error_msg.format(url))
+            except Exception as e:
+                raise e
+    else:
+        with warnings.catch_warnings():
+            # Perhaps use requests iter_content/iter_lines as below to read large CSVs so progress can be shown
+            # Progress bar is not used because TqdmReader object has no attribute 'seek' and would need to be modified to not have a newline operator
+            warnings.simplefilter("ignore", category=pd.errors.DtypeWarning)
+            table = pd.read_excel(url, nrows=limit, engine='openpyxl')                
+
+    if limit!=None and len(table) > limit:
+        table = table.head(limit)
+    
+    table = filter_dataframe(table, date_field=date_field, year_filter=year_filter, 
+        agency_field=agency_field, agency=agency)
+        
+    return table        
+
+def filter_dataframe(df, date_field=None, year_filter=None, agency_field=None, agency=None):
```

Comment:
I don't think it ever runs because filter_dataframe redefined later in this file. Is there any reason that you couldn't just use the original filter_dataframe function?

Reply:
made the change